src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1002,6 +1002,14 @@ void VM_Version::get_processor_features() {
     }
   }
 
+  // APX support code currently requires UseAPX > 2. This may change in the future.
+  if (UseAPX && (UseAVX < 3)) {
+    if (!FLAG_IS_DEFAULT(UseAPX)) {
+        warning("UseAPX is only available when UseAVX > 2");
```

Comment:
```suggestion
        warning("UseAPX is only available when UseAVX > 2. Disabling UseAPX.");
```

Reply:
This would tell the user what we are doing, just like with the UseAVX flag.

Reply:
Thank you, done.